linked_list.rb

```diff
@@ -0,0 +1,138 @@
+load "node_linked_list.rb"
+
+class Linked_list
+	attr_accessor :head_node
+
+	def search_element(value)
+		result = false
+		current_node = @head_node
+
+		while current_node != nil
+			if(current_node.data == value)
+				result = true
+				break
+			end
+			current_node = current_node.next
+		end
+		return result
+	end
+
+	def delete_node(value)
+		current_node = @head_node
+
+		if(current_node != nil && current_node.data == value)
+			@head_node = current_node.next
+			return
+		else
+			prev = nil
+			while (current_node != nil && current_node.data != value)
+				prev = current_node
+				current_node = current_node.next
+			end
+
+			if current_node == nil
+				return
+			end
+
+			prev.next = current_node.next
+		end
+	end
+
+	def create_linked_list(input)
+		@head_node = nil
+
+		input.each do |n|
+			insert_node(n)
+		end
+	end
+
+	def insert_node(value)
+		new_node = NodeLinkedList.new(value)
+		current_node = @head_node
+
+		if(current_node == nil)
+			@head_node = new_node
+			return
+		end
+
+		while current_node.next != nil
+			current_node = current_node.next
+		end
+
+		current_node.next = new_node
+	end
+
+	def reverse_linked_list()
+		current_node = @head_node
+		prev_node = nil
+		next_node = nil
+
+		while current_node != nil
+			new_node = current_node.next
+			current_node.next = prev_node
+			prev_node = current_node
+			current_node = new_node
+		end
+
+		@head_node = prev_node
+	end
+
+	def print_linked_list()
+		current_node = @head_node
+		while current_node != nil
+			puts current_node.data
+			current_node = current_node.next
+		end
+	end
+
+	def perform_task()
+		different_task = 
+				{quit: 0,
+				search_element: 1,
+				delete_node: 2,
+				insert_node: 3,
+				reverse_ll: 4,
+				print_ll: 5,
+			 	}
+
+		puts "#{different_task.fetch(:quit)}: Quit",
```

Comment:
you can also use `different_task[:quit]`. This doesn't raise error
.fetch  can raise error if key doesn't exist, so use it properly